### PR TITLE
List of current changes:

### PR DESCRIPTION
**Pull request by IhorKryviuk**
1.	Add transaction support for purchasing product operation.
2.	Change  url '/' handler from HomePgeController to LoginController.
3.	Add HistoryDTO.
4.	Implement Builder Pattern for the common.model.Cart, common.model.dto.CartDTO and common.model.dto.HistoryDTO classes.
5.	Improve mvcapp.controller.CartController getCarts method test.
6.	Add tests for the mvcapp.service.CartService, restservice.controller.CartController and restservice.service.CartService classes.
7.	Change mvcapp.controller.HistoryController deleteHistory method mapping from @GetMapping("/delete") to @GetMapping("/delete/{historyId}").
8.	Move modalDeleteConfirm dialog window to templates.
9.	Change references to appropriate pages in mvc\…\templates\header.jsp and mvc\…\ views\home.jsp from relative to the current directory (like "../home") to relative to the webapps directory (like "/home").

### CODE REVIEW

File: common-modules/src/main/java/com/softserve/if072/common/model/Cart.java [hunk 1]
```diff
@@ -66,4 +66,80 @@ public String toString() {
                 product.getUnit().getName() +
                 "\n}";
     }
+
+    public static class Builder {
```
Comment (kucherif) [inline common-modules/src/main/java/com/softserve/if072/common/model/Cart.java:70]: Better to move this to the test classes. I mean to put under src/test/java/com/softserve/if072/common/model/CartBuilder.java

File: common-modules/src/main/java/com/softserve/if072/common/model/dto/CartDTO.java [hunk 2]
```diff
@@ -72,4 +72,49 @@ public String toString() {
                 ", amount: " + amount +
                 "}";
     }
+
+    public static class Builder {
```
Comment (kucherif) [inline common-modules/src/main/java/com/softserve/if072/common/model/dto/CartDTO.java:76]: this is similar. code for tests shouldn't be with application code

File: common-modules/src/main/java/com/softserve/if072/common/model/Cart.java (hunk 1 above)
Comment (IhorKryviuk) [reply common-modules/src/main/java/com/softserve/if072/common/model/Cart.java:70]: I'm not sure if I use such builders only in test cases in the future.

Comment (IhorKryviuk) [reply common-modules/src/main/java/com/softserve/if072/common/model/Cart.java:70]: If I don't have to use it, I'll move it to the appropriate path.

**Pull request merged**